src/core/stream/representation/representation_stream.ts

```diff
@@ -377,13 +382,96 @@ export default function RepresentationStream<T>({
+                    acc.nonInterpretedEvents.push(val);
+                  }
+                  return acc;
+                }, { manifestRefreshEvents: [] as IInbandEvent[],
+                     nonInterpretedEvents: [] as IInbandEvent[] });
+              const manifestRefresh$ = clock$.pipe(
+                take(1),
+                mergeMap(({ position }) => {
+                  if (manifestRefreshEvents.length === 0) {
+                    return EMPTY;
+                  }
+                  let minManifestExpiration: undefined | number;
+                  const len = manifestRefreshEvents.length;
+                  for (let i = 0; i < len; i++) {
+                    const manifestRefreshEvent = manifestRefreshEvents[i];
+                    const currentManifestPublishTime = content.manifest.publishTime;
+                    const { messageData } = manifestRefreshEvent;
+                    const strPublishTime = utf8ToStr(messageData);
+                    const eventManifestPublishTime = Date.parse(strPublishTime);
+                    if (currentManifestPublishTime === undefined ||
+                        eventManifestPublishTime === undefined ||
+                        isNaN(eventManifestPublishTime) ||
+                        // DASH-if 4.3 tells (4.5.2.1) :
+                        // "The media presentation time beyond the event time (indicated
+                        // time by presentation_time_delta) is correctly described only
+                        // by MPDs with publish time greater than indicated value in the
+                        // message_data field."
+                        //
+                        // Here, if the current manifest has its publish time superior
+                        // to event manifest publish time, then the manifest does not need
+                        // to be updated
+                        eventManifestPublishTime < currentManifestPublishTime) {
+                      break;
+                    }
+                    const { timescale, presentationTimeDelta } = manifestRefreshEvent;
+                    const eventPresentationTime =
+                      (evt.segment.time / evt.segment.timescale) +
+                      (presentationTimeDelta / timescale);
+                    minManifestExpiration =
+                      minManifestExpiration === undefined ?
+                        eventPresentationTime :
+                        Math.min(minManifestExpiration,
+                                 eventPresentationTime);
+                  }
+                  if (minManifestExpiration === undefined) {
+                    return EMPTY;
+                  }
+                  // redefine manifest expiration time, as typescript does not understand
+                  // that it can't be undefined when using it in the getDelay callback
+                  const manifestExpirationTime = minManifestExpiration;
+                  const delayComputingTime = performance.now();
```

Comment:
99% of that looks to me like it doesn't have its place in the Stream.
Shouldn't we move it either to the `transports` code or the `Manifest` code?